### PR TITLE
feat(week-view): 5 days semi-weeks

### PR DESCRIPTION
**Pull request by azachar**
Hello,
I have implemented a new feature that allows you to show only part of the week.
image

It would be great if you could help me to review the new code. 

Also, I am not sure how to fix one failing test ( to be honest that view I do not need, so I left the investigation over). Perhaps you could have an idea how to fix it. I guess it would be for you super easy.

Anyway, thanks for your feedback!

Cheers,
Andrej

### CODE REVIEW

File: src/less/grid.less [hunk 1]
```diff
@@ -40,82 +42,9 @@ mwl-calendar {
   .cal-row-fluid .controls-row [class*="cal-cell"] + [class*="cal-cell"] {
     margin-left: 0%;
   }
-  .cal-row-fluid .cal-cell7 {
-    width: 100%;
-    *width: 99.94669509594883%;
-  }
-  .cal-row-fluid .cal-cell6 {
-    width: 85.71428571428571%;
-    *width: 85.66098081023453%;
```
Comment (azachar) [inline src/less/grid.less]: From where this constant come from? THX.

File: src/less/grid-mixin.less [hunk 2]
```diff
@@ -0,0 +1,33 @@
+.gridForWeekOf(@days) {
+  @oneDay: 100% / @days;
+  @oneDayIE: 0.9993781095 * @oneDay;
```
Comment (azachar) [inline src/less/grid-mixin.less]: I hope I set the constant according to your manual calculations.

**Comment by mattlewis92** [comment]
Thanks this looks good! I don't have time to properly review it right now, however there are a couple of things I can see that you can work on now:
- [ ] please remove the dist files from the commit, these get generated automatically on release
- [ ] please can you remove the event editor from the new demo - I prefer to keep feature demos focused on just that particular feature
- [ ] the `calendar-utils` lib already supports excluding days from the week view - you can pass an array of day indices to hide from the week view e.g. `getWeekView({..., excluded: [0, 6]})` will remove all weekends from the view - please can you change the API do follow this approach instead of just passing a count of days to exclude - this way people will be able to choose the days they want to hide and leads to a more flexible API. You can see an example of how this works for the angular2 version of this calendar [here](https://mattlewis92.github.io/angular-calendar/#/exclude-days)

**Comment by azachar** [comment]
Hey @mattlewis92, 

Thanks for your message. 


ad 1)  I need to somehow used within my project, so until we merge this PR, I will keep it there. (but yes I agree with you:)

ad 2) Sure, will remove it. (was there just temporary)

ad 3) calendar-utils support excluding, that is great, but it is a quite a messy to use it for this feature (initially I used but the algorithm to choose excluded days became very smelly). 
Since my feature is not about hiding, it is about making the calendar available on devices that have a small screen (like a mobile phone for instance). So one semi week of 3 days starts for example on T(uesday), then continue to W(ednesay) and finish up with F(riday).
The next semi week is S,S,M, then T,W,F..  and still, it makes sense to filter out weekends.

So, I wound prefer to keep approaches transparent and not to reuse them. They have in my opinion different meanings. Do you see my point? And also, this should be in theory faster (instead of excluding, we are just iterating less days..)


Anyway, look forward hearing from you!

Cheers,
Andrej

**Comment by azachar** [comment]
ad 2) removed :)

**Comment by DarshanVelagaleti** [comment]
can any one please help me to Hide Previous dates and after 60 days Dates on Calander

**Comment by mattlewis92** [comment]
Sorry about the late reply on this one, I don't really work on angular 1.x stuff anymore. Thanks to the work by @santam85 this is now possible in 0.30.0 🎉

**Pull request closed without merging**